apps/oracledb.go

```diff
@@ -78,6 +79,9 @@ func (r MetricsReceiverOracleDB) Pipelines(_ context.Context) ([]otel.ReceiverPi
 	auth := url.QueryEscape(r.Username)
 	secretPassword := r.Password.SecretValue()
 	if len(secretPassword) > 0 {
+		if strings.Contains(secretPassword, "${googlesecretmanager") {
```

Comment:
Nit. Wouldn't we want `secretPassword`  to start with `${googlesecretmanager`? (i.e. `HasPrefix`)

Reply:
Currently, we are trying to prevent users from using the secret provider in OracleDB-related configuration entries. 

This is because the password will be URI-escaped before being passed to the OTel collector, which results in the provider identifier being altered and consequently unable to map back to a secret entry in Google Secret Manager. 

We will remove this check later when we implement the fix in the sqlquery receiver.

Reply:
Here's the upstream issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/40117

Reply:
Based on my previous comment, I don't think deciding whether to implement escaping in a new provider or sql receiver should be blocking for the Secret Manager integration.

Reply:
Also copied from the upstream issue:

> This isn't a question of the "ideal" solution - the proposed solution here can not work. Once the string is already concatenated into a single URL, it's no longer possible to know which part of that string needs encoding applied to it.

Also, the exact same problem applies to environment variable substitution, not just `googlesecretmanager`, so if you want to report a configuration error until this is supported upstream, you should be checking for `${` and not just `${googlesecretmanager`

Reply:
That is a good point. This would also not work with env variable substitution. Maybe we do a regex match for  something like`${.*}`

Let's also change the error message to say that this integration does not support confmap providers.

Reply:
Done